### PR TITLE
util: avoid column allocator reuse the column hold huge memory

### PR DESCRIPTION
**Pull request by wshwsh12**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/31981
close https://github.com/pingcap/tidb/issues/30880

Problem Summary:
Chunk-rpc will use rpc message to construct chunk directly, which will hold the whole memory.
We should release them after the sql finished.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
Fix the problem of tidb oom when exporting data using chunk rpc
```

### CODE REVIEW

**Review by tiancaiamao — APPROVED** [review]

File: util/chunk/codec.go [hunk 1]
```diff
@@ -140,6 +140,7 @@ func (c *Codec) decodeColumn(buffer []byte, col *Column, ordinal int) (remained
 
 	// decode data.
 	col.data = buffer[:numDataBytes:numDataBytes]
+	// the column reference the data of grpc, so need set col.avoidReusing to avoid reusing it.
```
Comment (XuHuaiyu) [inline util/chunk/codec.go:143]: The column reference the data of the grpc response, the memory of the grpc message cannot be GCed if we reuse this column. Thus we set \`avoidResuing\` to true.

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**